### PR TITLE
Convert the Logical to Physical map to a visitor

### PR DESCRIPTION
**Pull request by Fokko**
I noticed that the FixedType was missing.

### CODE REVIEW

File: pyiceberg/io/pyarrow.py [hunk 1]
```diff
@@ -1099,21 +1099,70 @@ def map_value_partner(self, partner_map: Optional[pa.Array]) -> Optional[pa.Arra
...
-    StringType(): "BYTE_ARRAY",
-    UUIDType(): "FIXED_LEN_BYTE_ARRAY",
-    BinaryType(): "BYTE_ARRAY",
-}
-_PHYSICAL_TYPES = set(_PRIMITIVE_TO_PHYSICAL.values()).union({"INT96"})
+def _primitive_to_phyisical(iceberg_type: PrimitiveType) -> str:
+    return visit(iceberg_type, _PRIMITIVE_TO_PHYISCAL_TYPE_VISITOR)
+
+
+class PrimitiveToPyhsicalType(SchemaVisitorPerPrimitiveType[str]):
```
Comment (rdblue) [inline pyiceberg/io/pyarrow.py:1106]: Typo: `Pyhsical` (unless maybe that's how it's spelled in the python community)

File: pyiceberg/io/pyarrow.py [hunk 2]
```diff
@@ -1099,21 +1099,70 @@ def map_value_partner(self, partner_map: Optional[pa.Array]) -> Optional[pa.Arra
...
+        raise ValueError(f"Expected primitive-type, got: {list_type}")
+
+    def map(self, map_type: MapType, key_result: str, value_result: str) -> str:
+        raise ValueError(f"Expected primitive-type, got: {map_type}")
+
+    def visit_fixed(self, fixed_type: FixedType) -> str:
+        return "BYTE_ARRAY"
+
+    def visit_decimal(self, decimal_type: DecimalType) -> str:
+        raise ValueError("unknown")
```
Comment (rdblue) [inline pyiceberg/io/pyarrow.py:1126]: We could convert to FIXED_LEN_BYTE_ARRAY or INT64 or INT32 based on the precision, right?

**Review by rdblue — APPROVED** [review]
Looks good overall, but I'd fix the typo before merging.

File: pyiceberg/io/pyarrow.py (hunk 1 above)
Comment (Fokko) [reply pyiceberg/io/pyarrow.py:1106]: 🤣  Good old typo

File: pyiceberg/io/pyarrow.py (hunk 2 above)
Comment (Fokko) [reply pyiceberg/io/pyarrow.py:1126]: I would say `FIXED_LEN_BYTE_ARRAY`. This visitor is just to double-check that we get the type that we expect.

**Comment by Fokko** [comment]
Thanks @rdblue for the quick review! 🙌

**Pull request merged**